WIP: Baselayer refactor

acrellin opened the pull request:
To-do (partial):
- [x] Move `app_server` to `baselayer`
- [x] Move `nginx` and `supervisor` config files to `baselayer`
- [x] Move `services` to `baselayer`
- [x] Move some modules in `tools` to `baselayer`
- [ ] Add install script
...

Closes #150

acrellin committed:
Move BaseHandler and AccessError to baselayer

acrellin committed:
Move BaseModel, json_util, model_util to baselayer

acrellin committed:
Move User model to baselayer; fix broken imports

acrellin committed:
Move psa module to baselayer

acrellin committed:
Separately create baselayer and cesium_web models

acrellin committed:
Move flow to baselayer

pep8speaks commented:
Hello @acrellin! Thanks for updating the PR.



 - In the file [`baselayer/app/app_server.py`](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/app_server.py), following are the PEP8 issues :

> [Line 20:1](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/app_server.py#L20): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented

 - In the file [`baselayer/app/config.py`](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/config.py), following are the PEP8 issues :

> [Line 50:46](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/config.py#L50): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 53:37](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/config.py#L53): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 54:37](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/config.py#L54): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 58:33](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/config.py#L58): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 59:33](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/config.py#L59): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator











 - In the file [`baselayer/app/model_util.py`](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/model_util.py), following are the PEP8 issues :

> [Line 17:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/model_util.py#L17): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare except'
> [Line 26:17](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/model_util.py#L26): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 27:17](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/model_util.py#L27): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 27:17](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/app/model_util.py#L27): [W503](https://duckduckgo.com/?q=pep8%20W503) line break before binary operator





 - In the file [`baselayer/example_app/app/app_server.py`](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py), following are the PEP8 issues :

> [Line 31:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L31): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 32:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L32): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 33:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L33): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 34:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L34): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 35:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L35): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 37:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L37): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 38:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L38): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 39:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L39): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 40:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L40): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 41:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L41): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 42:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L42): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 43:5](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/example_app/app/app_server.py#L43): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment









 - In the file [`baselayer/tools/supervisor_status.py`](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/tools/supervisor_status.py), following are the PEP8 issues :

> [Line 10:1](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/tools/supervisor_status.py#L10): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 11:81](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/baselayer/tools/supervisor_status.py#L11): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 80 characters)

























 - In the file [`cesium_app/tests/conftest.py`](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/cesium_app/tests/conftest.py), following are the PEP8 issues :

> [Line 16:23](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/cesium_app/tests/conftest.py#L16): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 17:23](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/cesium_app/tests/conftest.py#L17): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator

 - In the file [`tools/test_frontend.py`](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/tools/test_frontend.py), following are the PEP8 issues :

> [Line 12:1](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/tools/test_frontend.py#L12): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 18:1](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/tools/test_frontend.py#L18): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 19:1](https://github.com/cesium-ml/cesium_web/blob/f33e4136df07d0759d59ecd1e529a0b20f913d05/tools/test_frontend.py#L19): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file



##### Comment last updated on June 15, 2017 at 17:32 Hours UTC

acrellin committed:
Very WIP commit of current state

stefanv committed:
Further refactoring of Cesium -> baselayer

acrellin committed:
Update Makefile target paths

acrellin committed:
Add db_init.py and modify Makefile and db_init.sh

acrellin committed:
Read cesium_web-level config file by default; further refactor to get working

acrellin committed:
Update Makefile and paths

bnaul committed:
supervisor testing conf

bnaul committed:
Change supervisor status call

bnaul committed:
Set PYTHONPATH in db init

bnaul committed:
Remove unused settings

acrellin committed:
Update config file loading

bnaul committed:
Removed unused app logic

stefanv commented on baselayer/app/app_server.py:
```diff
@@ -0,0 +1,101 @@
+    'login_url': '/',
+
+    # Python Social Auth configuration
+    'SOCIAL_AUTH_USER_MODEL': 'cesium_app.models.User',
```
How do we customize this for each app?

stefanv commented on baselayer/app/handlers_and_settings.py:
```diff
@@ -0,0 +1,58 @@
+    'login_url': '/',
+
+    # Python Social Auth configuration
+    'SOCIAL_AUTH_USER_MODEL': 'cesium_app.models.User',
```
Duplicate?

stefanv commented on baselayer/conf/supervisor/testing.conf:
```diff
@@ -0,0 +1,8 @@
+files = common.conf
+
+[program:app]
+command=/usr/bin/env python baselayer/services/app.py --config cesium.yaml.example --config _cesium_test.yaml
```
Customized for cesium?

stefanv commented on baselayer/example_app/static/js/main.jsx:
```diff
@@ -0,0 +1,426 @@
+import React from 'react';
```
Example app should be minimal, not full Cesium.

stefanv commented on baselayer/static/js/MessageHandler.jsx:
```diff
@@ -0,0 +1,41 @@
+import * as Action from './actions';
```
This is Cesium specific.

stefanv commented on baselayer/tools/db_clear.py:
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
+
+from cesium_app import app_server
```
Cesium specific.

stefanv reviewed (commented):
Overall looking great, thanks for all the hard work @acrellin and @bnaul!

We need to be careful not to place any Cesium-specifics in Baselayer—I've marked these in a few spots.  But there aren't many such cases, so I believe we're close.

stefanv commented on baselayer/Makefile:
```diff
@@ -0,0 +1,86 @@
+SHELL = /bin/bash
```
Lots more cesium specific targets.

bnaul committed:
Clean up config loading logic

bnaul committed:
Move baselayer config into example app

bnaul committed:
Use baselayer User model

bnaul committed:
Rename config files

stefanv committed:
Only load config.yaml, not *.yaml

- When the user specifies a config file, load

  1. baselayer/config.yaml.example (baselayer defaults)
  2. config.yaml.example (defaults)
  3. User-specified configs

- Otherwise, load

  1. baselayer/config.yaml.example (baselayer defaults)
  2. config.yaml.example (defaults)
  3. baselayer/config.yaml (baselayer config)
  3. config.yaml (app config)

stefanv committed:
Move Cesium-specific message handling into separate module

stefanv committed:
De-duplicate baselayer and cesium

- Move all Redux components into separate directories
- Ensure files only exist in either cesium or baselayer, but not both
- Rename non-JSX files to have .js extension
- Refactor MessageHandler to first handle baselayer messages, and to then
  pass control to the user application

stefanv committed:
Remove mentions of skyportal
